compiler/packages/babel-plugin-react-compiler/src/Inference/DropManualMemoization.ts

```diff
@@ -476,3 +484,46 @@ export function dropManualMemoization(func: HIRFunction): void {
     }
   }
 }
+
+function findOptionalPlaces(fn: HIRFunction): Set<IdentifierId> {
+  const optionals = new Set<IdentifierId>();
+  for (const [, block] of fn.body.blocks) {
+    if (block.terminal.kind === 'optional') {
+      const optionalTerminal = block.terminal;
+      let testBlock = fn.body.blocks.get(block.terminal.test)!;
+      loop: while (true) {
+        const terminal = testBlock.terminal;
+        switch (terminal.kind) {
+          case 'branch': {
+            if (terminal.fallthrough === optionalTerminal.fallthrough) {
+              // found it
+              const consequent = fn.body.blocks.get(terminal.consequent)!;
+              const last = consequent.instructions.at(-1);
+              if (last !== undefined && last.value.kind === 'StoreLocal') {
+                optionals.add(last.value.value.identifier.id);
+              }
+              break loop;
+            } else {
+              testBlock = fn.body.blocks.get(terminal.fallthrough)!;
+            }
+            break;
+          }
+          case 'optional':
+          case 'logical':
+          case 'sequence':
+          case 'ternary': {
```

Comment:
Ah I see, I'm guessing that this is to support something like `useMemo(..., [(a ?? b)?.c])`. Not technically necessary for `DropManualMemo` as we're only tracking named variables, but this makes sense to have in a utility function

Reply:
yup exactly